### PR TITLE
Correct Missing Type Annotations

### LINKED ISSUE #3: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_aw9523.py:70
- [ ] adafruit_aw9523.py:85
- [ ] adafruit_aw9523.py:108
- [ ] adafruit_aw9523.py:122
- [ ] adafruit_aw9523.py:131
- [ ] adafruit_aw9523.py:140
- [ ] adafruit_aw9523.py:159
- [ ] adafruit_aw9523.py:163
- [ ] adafruit_aw9523.py:167
- [ ] adafruit_aw9523.py:181
- [ ] adafruit_aw9523.py:191
- [ ] adafruit_aw9523.py:198
- [ ] adafruit_aw9523.py:217
- [ ] adafruit_aw9523.py:233
- [ ] adafruit_aw9523.py:250

### PR DESCRIPTION
**Pull request by tcfranks**
resolves #3 
submitted for review / comment

### CODE REVIEW

File: adafruit_aw9523.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # SPDX-FileCopyrightText: Copyright (c) 2021 ladyada for Adafruit
+# SPDX-FileCopyrightText: Copyright (c) 2021 ladyada for Adafruit
```
Comment (FoamyGuy) [inline adafruit_aw9523.py]: Did this line get duplicated accidentally? I think we don't need two copies of this copyright.

File: adafruit_aw9523.py [hunk 2]
```diff
@@ -124,30 +133,30 @@ def get_pin(self, pin):
...
 
     @property
-    def directions(self):
+    def directions(self) -> int:
         """Direction is output if bit mask is 1, input if bit is 0"""
         return ~self._directions & 0xFFFF
 
     @directions.setter
-    def directions(self, dirs):
+    def directions(self, dirs) -> None:
```
Comment (FoamyGuy) [inline adafruit_aw9523.py]: `dirs` can be `int` I think.

File: adafruit_aw9523.py [hunk 3]
```diff
@@ -124,30 +133,30 @@ def get_pin(self, pin):
...
 
     @property
-    def LED_modes(self):
+    def LED_modes(self) -> int:
         """Pin is set up for constant current mode if bit mask is 1"""
         return ~self._LED_modes & 0xFFFF
 
     @LED_modes.setter
-    def LED_modes(self, modes):
+    def LED_modes(self, modes) -> None:
```
Comment (FoamyGuy) [inline adafruit_aw9523.py]: `modes` can be `int` I think

**Review by FoamyGuy — CHANGES_REQUESTED** [review]
Thanks for working on this @tcfranks! 

I have a question and few suggested changes in places where some arguments didn't get types yet.

File: adafruit_aw9523.py (hunk 1 above)
Comment (tcfranks) [reply adafruit_aw9523.py]: oops - ctrl - D in pycharm duplicates the current line I think - it sometimes gets accidently pressed but I must have missed this one

File: adafruit_aw9523.py (hunk 3 above)
Comment (tcfranks) [reply adafruit_aw9523.py]: wow, I missed both setters!  my bad.

**Comment by tcfranks** [comment]
submitted with requested changes but forgot to retitle the commit

**Review by FoamyGuy — APPROVED** [review]
Looks good to me. Thanks again!

**Pull request merged**